### DISCUSSION TITLE
Highlight is reverting to default

### DISCUSSION THREAD

**Discussion by EliezerAmbion** [nvim-treesitter/nvim-treesitter#5168] [Q&A]
Hi! I am new to neovim. I don't know if my setup of treesitter is correct.

The issue is I want to change the highlight @include (and some other highlights).
It is working upon start-up. But when I change something on the file, the colors will revert back to default.

My config:
```lua
return {
  'nvim-treesitter/nvim-treesitter',
  build = ':TSUpdate',

  opts = {
    ensure_installed = { 'dart', 'javascript', 'typescript' },
    sync_install = false,
    auto_install = true,
    highlight = {
      enable = true,
      additional_vim_regex_highlighting = false,
    },
  },

  config = function(_, opts)
    -- change highlight color of include
    vim.api.nvim_set_hl(0, "@include", { link = 'Title' })

    require('nvim-treesitter.configs').setup(opts)
  end
}
```

Is there something that I did wrong?
I don't know if the issue persists in other languages but I am using dart/ flutter.

Upon start-up (correct highlight)
![Screenshot 2023-07-29 at 1 29 26 AM](https://github.com/nvim-treesitter/nvim-treesitter/assets/57052385/4fee60b0-116d-44a5-9389-4cb157903e78)

Reverted back to default
![Screenshot 2023-07-29 at 1 30 45 AM](https://github.com/nvim-treesitter/nvim-treesitter/assets/57052385/802d163a-41b6-4428-a1f8-820e1423ff7d)

**Comment by clason** [comment]
This is probably your LSP serving up semantic tokens. You can tell by calling `:Inspect` on the "wrong" highlight.
